engine/common.ftl

```diff
@@ -1,5 +1,59 @@
 [#ftl]
 
+[#-- DateTime functions --]
+[#function DateTimeAdd dateTime unit amount]
+    [#switch unit]
+        [#case "dd"]
+            [#local offset=1000*60*60*24]
+            [#break]
+        [#case "hh"]
+            [#local offset=1000*60*60]
+            [#break]
+        [#case "mm"]
+            [#local offset=1000*60]
+            [#break]
+    [/#switch]
+    [#local retval = (dateTime?datetime?long + amount?long*offset?long)?number_to_datetime]
+    [#return retval ]
+[/#function]
+
+[#function ShowDateTime dateTime format timezone="UTC" ]
+    [#local old_time_zone = .time_zone]
+    [#setting time_zone = timezone]
+    [#local retval = dateTime?datetime?string(format)]
+
+    [#setting time_zone = old_time_zone]
+    [#return retval ]
+[/#function]
+
+[#function DayOfWeek2DateTime dayOfWeek timeofDay="00:00" timeZone="UTC"]
+    [#switch dayOfWeek]
+        [#case "Monday"]
+            [#local startTime = "05-JUL-2021 "+timeofDay+":00 "+timeZone ]
+            [#break]
+        [#case "Tuesday"]
+            [#local startTime = "06-JUL-2021 "+timeofDay+":00 "+timeZone ]
+            [#break]
+        [#case "Wednesday"]
+            [#local startTime = "07-JUL-2021 "+timeofDay+":00 "+timeZone ]
+            [#break]
+        [#case "Thursday"]
+            [#local startTime = "08-JUL-2021 "+timeofDay+":00 "+timeZone ]
+            [#break]
+        [#case "Friday"]
+            [#local startTime = "09-JUL-2021 "+timeofDay+":00 "+timeZone ]
+            [#break]
+        [#case "Saturday"]
+            [#local startTime = "10-JUL-2021 "+timeofDay+":00 "+timeZone ]
+            [#break]
+        [#case "Sunday"]
+            [#local startTime = "11-JUL-2021 "+timeofDay+":00 "+timeZone ]
+            [#break]
```

Comment:
What are these values for with the current date? Could we instead use ?now which gets the current time?

Reply:
These are to convert a "day" to a date which is sufficiently far away from any leap years etc to allow for timezone manipulations and date arithmetic (eg Saturday 01:00AEST to become Friday 15:00UTC)

Reply:
I think we need a comment on this as to why we have added fixed dates, its a bit weird to see specific dates in the code is all.

Reply:
Cool, done